features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/prerequisite/SessionReplayRequirementTest.kt

```diff
@@ -0,0 +1,261 @@
+            internalLogger = mockInternalLogger,
+            runnable = {
+                val isOnTheSameLooper = outsideLooper == Looper.myLooper()
+                completableFuture.complete(isOnTheSameLooper)
+            }
+        )
+        completableFuture.thenAccept {
+            assertThat(it).isFalse()
+        }.orTimeout(100, TimeUnit.MILLISECONDS).join()
+    }
+
+    @Test
+    fun `M send telemetry W task is not executed`() {
+        // Given
+        val testedSystemRequirementConfiguration = SessionReplayRequirement.Builder()
+            .setSystemRequirementConfiguration(fakeSystemRequirementConfiguration)
+            .addChecker(mockRefuseChecker)
+            .build()
+        val stubRunnable = StubRunnable()
+
+        // When
+        testedSystemRequirementConfiguration.runIfMeetRequirements(
+            internalLogger = mockInternalLogger,
+            runnable = {
+                stubRunnable.run()
+            }
+        )
+
+        Thread.sleep(100)
+
+        // Then
+        val captorLevel = argumentCaptor<InternalLogger.Level>()
+        val captorTargets = argumentCaptor<List<InternalLogger.Target>>()
+        val captorThrowable = argumentCaptor<Throwable>()
+        val captorMessageBuilder = argumentCaptor<() -> String>()
+        val captorMap = argumentCaptor<Map<String, Any?>>()
+
+        verify(mockInternalLogger).log(
+            level = captorLevel.capture(),
+            targets = captorTargets.capture(),
+            messageBuilder = captorMessageBuilder.capture(),
+            throwable = captorThrowable.capture(),
+            onlyOnce = any(),
+            additionalProperties = captorMap.capture()
+        )
+        assertThat(captorLevel.firstValue).isEqualTo(InternalLogger.Level.INFO)
+        assertThat(captorTargets.firstValue).isEqualTo(
+            listOf(
+                InternalLogger.Target.TELEMETRY,
+                InternalLogger.Target.USER
+            )
+        )
+        assertThat(captorMessageBuilder.firstValue.invoke()).isEqualTo(
+            "Session replay is disabled because the system doesn't meet the minimum SDK requirements"
+        )
+        val additionalProperties = captorMap.firstValue as Map<*, *>
+        val deviceStatsMap = additionalProperties["device_stats"] as Map<*, *>
+        assertThat(deviceStatsMap[mockRefuseChecker.name()]).isEqualTo(fakeCheckerValue)
+    }
+
+    private class StubRunnable : Runnable {
+        var isExecuted = false
+        var isExecutedOnMainThread = false
+        override fun run() {
+            isExecutedOnMainThread = Looper.getMainLooper() == Looper.myLooper()
+            isExecuted = true
+        }
+    }
+
+    companion object {
+        private val mainLooper = MainLooperTestConfiguration()
+
+        @TestConfigurationsProvider
+        @JvmStatic
+        fun getTestConfigurations(): List<TestConfiguration> {
+            return listOf(mainLooper)
+        }
+    }
```

Comment:
We don't need a `Looper`. This is a concept from Android and this test is running on JVM, not on Android. You can simply use existing Thread API to spin a new thread and block the existing one until the execution is not completed for the spinned one.

Reply:
Updated, now the unit test just simply check if the thread where the runnable is executed is the same as the one of the call site